### ISSUE TITLE
Map server returns PNG encoding error

### ISSUE THREAD

**Issue by lost-geographer** [lovasoa/dezoomify-rs#61]
Hello everybody,

I am trying to download a topographical map accessible from here : http://tiles.webmapp.it/apuane_25k_2018_288dpi/apuane_25k_2018_288dpi/#apuane_25k_2018_288dpi/ol3. I thought that something simple like this would work :

`dezoomify-rs -r 0 http://tiles.webmapp.it/apuane_25k_2018_288dpi/apuane_25k_2018_288dpi/16/{{X}}/{{Y}}.png`

Or something more complex like this (following the example shown [here](https://github.com/lovasoa/dezoomify/issues/380#issuecomment-612965275)) :

`dezoomify-rs --retries 0 "http://tiles.webmapp.it/apuane_25k_2018_288dpi/apuane_25k_2018_288dpi/wmt/wmts?&crs=EPSG:3857&dpiMode=7&format=image/png&layers=apuane_25k_2018_288dpi&styles=default&tileMatrixSet=GoogleMapsCompatible&tilematrix=16&tilerow={{X}}&tilecol={{Y}}"`

But in each case I got this error : `ERROR PNG encoding error: Zero width not allowed`.

Do you have any idea ?

**Comment by lovasoa** [comment]
It looks like your URL template doesnt lead to valid image links, at least for x=0 and y=0 : 

http://tiles.webmapp.it/apuane_25k_2018_288dpi/apuane_25k_2018_288dpi/16/0/0.png

**Comment by lovasoa** [comment]
Once you figure out which are the min and max allowed values for x and y, you can use the [custom yaml dezoomer](https://github.com/lovasoa/dezoomify-rs#Custom-yaml)

**Comment by lovasoa** [comment]
I am going to close this specific issue, and create a new one concerning the support of WMTS images in dezoomify-rs. Supporting WMTS would give full support for this image.

**Comment by lost-geographer** [comment]
For the record : I did it ! Using the following command `dezoomify-rs -r 0 tiles.yaml result.jpg` and the YAML parameters below :

```
url_template: "http://tiles.webmapp.it/apuane_25k_2018_288dpi/apuane_25k_2018_288dpi/16/{{34587+x/256}}/{{23783+y/256}}.png"

# Each tile size is 256

# X/Y URL values at corners
# Top Right - Left   : 34587/23783 - 34682/23783
# Bottom Right - Left: 34587/23849 - 34682/23849

# Values start at the top right corner and finish at the bottom left one.
# X starts at 34587
# Y starts at 23783

# X URL value changes of 1 unit horizontally, so we have 34682−34587 = 95 tiles. The image width is 95x256 = 24320

# Y URL value changes of 1 unit vertically, so we have 23849−23783 = 66 tiles vertically. The image height : 66x256 = 16896

variables:
  - name: x
    from: 0   # Always from 0
    to: 24320 # Image width
    step: 256 # Tile width

  - name: y
    from: 0   # Always from 0
    to: 16896 # Image height
    step: 256 # Tile height
```

**Comment by lost-geographer** [comment]
> I am going to close this specific issue, and create a new one concerning the support of WMTS images in dezoomify-rs. Supporting WMTS would give full support for this image.

I think that it's not going to be easy because X/Y values vary following the geographical coordinates (sometimes it's not a simple +1 change). Anyway, thank you very much for your effort and patience @lovasoa !

**Comment by lovasoa** [comment]
First, congratulations and thanks for sharing your configuration file.  Did you have troubles understanding the documentation about the custom yaml format ? I sometimes get the feedback that it's not clear, but I'm not sure how to improve it. 

Second, I'm not sure I understand what you mean by "it's not a simple +1". Reading the documentation of the wmts format, I was under the impression that the image doesn't necessarily starts at 0,0 (as we've seen in this case), but the rest of the image is a simple rectangular matrix of tiles, in a given geographical coordinates system. Isn't it ?

**Comment by lost-geographer** [comment]
Regarding the YAML documentation, although it doesn't really fit my case, it is quite understandable. Having said that, I made a mistake on the first try by trying to put the tile numbers in the "from" and "to" fields. I hadn't realized that I had to calculate the dimensions of the image by myself (but the truth is that I hadn't read the instructions carefully :D).

> Second, I'm not sure I understand what you mean by "it's not a simple +1". Reading [the documentation of the wmts format](http://portal.opengeospatial.org/files/?artifact_id=35326), I was under the impression that the image doesn't necessarily starts at 0,0 (as we've seen in this case), but the rest of the image is a simple rectangular matrix of tiles, in a given geographical coordinates system. Isn't it ?

You are absolutely right I was misled by the URL formats I usually see in online maps, as in this case (https://lstg.eu/hike/etna/#12/37.7551/15.0063). The XY values of the page URL are indeed raw geographic coordinates, but the tiles are named differently, according to the total number of tiles composing the image (https://a.tile.hosted.thunderforest.com/komoot-2/12/2219/1583.png). The WMTS format seems to work in that way too.
